feat(service): add Opnform template

alexbaron-dev opened the pull request:
## Changes
- Add new template to deploy opnform app with coolify (see https://opnform.com/)

alexbaron-dev committed:
Create opnform.yaml

Add opnform.yaml as template to deploy OpnForm app

alexbaron-dev committed:
Add opnform logo into svgs directory

JhumanJ commented:
🔥

peaklabs-dev committed:
Merge branch 'next' into add-opnform-template

peaklabs-dev commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,232 @@
+# logo: svg/opnform.svg
+# port: 80
+
+x-shared-env: &shared-api-env
```
I think not all ENVs are needed all the time, so split them up, with only the ones needed for the container and some that are needed all the time.

alexbaron-dev replied:
I have kept here variables that are used by api, worker and scheduler and dispatched specific in dedicated service when applicable

peaklabs-dev commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,232 @@
+x-shared-env: &shared-api-env
+  APP_NAME: "OpnForm"
+  APP_ENV: production
+  APP_KEY: ${SERVICE_BASE64_APIKEY}
```
Please use the Coolify magic to automatically generate secrets like the app key.

alexbaron-dev replied:
APP_KEY is already a magic environment variables, and all secrets are generated with coolify. If not please tell me what's wrong.

peaklabs-dev commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,232 @@
+services:
+  opnform-api:
+    image: jhumanj/opnform-api:latest
+    volumes: &api-environment-volumes
```
Please just use plain volumes

alexbaron-dev replied:
Ok, I will duplicate the following line as api, worker and scheduler are using the same volume
    volumes:
      - api-storage:/usr/share/nginx/html/storage:rw

peaklabs-dev commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,232 @@
+
+  # The nginx reverse proxy.
+  # used for reverse proxying the API service and Web service.
+  nginx:
```
This should not be necessary as Coolify will handle the routing to the web application and API via magic ENVs. Please remove this if possible.

alexbaron-dev replied:
I have tried with traefik but unfortunatly I didn't success. If you have some suggestion with traefik I am open to update it later.

ShadowArcanist replied:
@peaklabs-dev I just tested this compose and it works fine for me. I tried removing the nginx but api routes like `/api/login` were not working and @alexbaron-dev mentioned they were not able to make this work without nginx, so I think we can keep the nginx

peaklabs-dev commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,232 @@
+      retries: 3
+      start_period: 40s
+      
+volumes:
```
This can be removed as Coolify will add this automatically.

alexbaron-dev committed:
Update opnform.yaml

- split environment variables per usage (api, worker and scheduler)
- remove volume anchor
- remove volumes definition

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 40s
```
Volumes section were missing
```suggestion
      start_period: 40s
volumes:
  redis-data:
  opnform-postgresql-data:
  api-storage:
```

JhumanJ replied:
This was a suggested change https://github.com/coollabsio/coolify/pull/5875/files/eef0e940e704f5413b309cc1d0099b26cd7267ee#r2104770506

JhumanJ commented:
Hey @ShadowArcanist @peaklabs-dev anything else for us to do here are we good? thanks!

JhumanJ commented:
Hey  @ShadowArcanist @peaklabs-dev any update?
thanks

ShadowArcanist commented:
Hey @JhumanJ I don’t have control over merging PRs in this repository, so I’m unable to provide a definitive update on the status of this one.

The decision ultimately rests with the Core Maintainers. I did notice that Peak, one of the maintainers, added a "Hold" label to the PR, which might suggest they’re currently prioritizing other tasks — but I can’t say for certain.

dropalltables commented:
I would also like an update on this!

Rodaviva29 commented:
This would be awesome, any ETA? Thanks guys!

AdamEXu commented:
This would be great!

alexbaron-dev commented:
Hello @peaklabs-dev, any update for this service ? 🙏🏽

Akount commented:
yes please !  @peaklabs-dev :) the one and only OpnForm is a must!

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+
+services:
+  opnform-api:
+    image: jhumanj/opnform-api:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+      start_period: 60s
+
+  api-worker:
+    image: jhumanj/opnform-api:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+      start_period: 30s
+  
+  api-scheduler:
+    image: jhumanj/opnform-api:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+      start_period: 70s # Allow time for first scheduled run and cache write
+
+  opnform-ui:
+    image: jhumanj/opnform-client:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+  # The nginx reverse proxy.
+  # used for reverse proxying the API service and Web service.
+  nginx:
+    image: nginx:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

alexbaron-dev committed:
Update Docker images to specific versions

alexbaron-dev committed:
Upgrade opnform API and UI images to version 1.10.0

alexbaron-dev committed:
Merge branch 'coollabsio:v4.x' into add-opnform-template

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,226 @@
+    environment:
+      POSTGRES_USER: ${SERVICE_USER_POSTGRESQL}
+      POSTGRES_PASSWORD: ${SERVICE_PASSWORD_POSTGRESQL}
+      POSTGRES_DB: ${POSTGRESQL_DATABASE:-opnform}
```
Please be consistent with formatting envs, on `opnform-ui` service you have env start with `-` and assigns using `=` but here and on some other services those formatting is not followed.

Check other service template we have on https://github.com/coollabsio/coolify/tree/v4.x/templates/compose to see how they have env formatted.

alexbaron-dev replied:
I have updated the file according to your recommandations

alexbaron-dev committed:
Update environment variable syntax in opnform.yaml for consistency

alexbaron-dev commented:
Hello, I have updated files as requested and add documentation (see https://github.com/coollabsio/coolify-docs/pull/401). I hope everything fine now :-)

yipfram commented:
It works for me even tho the generated url start by https://nginx-gggssc84s0515e4cc8o4c0.example.com)
You could maybe rename the `SERVICE_URL_NGINX` to `SERVICE_URL_OPNFORM` even if it is not in the envs of opnform container, to avoid confusion...

JhumanJ committed:
Update opnform.yaml to use version 1.10.1 for API and UI images, and correct APP_URL environment variable reference

JhumanJ committed:
Merge branch 'next' into add-opnform-template

JhumanJ commented:
Hey @yipfram thanks for approving. Issue still have waiting for changes label, can we please get this changed?
When we can expect this to be merged? 
Thanks
<img width="1188" height="160" alt="CleanShot 2025-11-08 at 13 09 20@2x" src="https://github.com/user-attachments/assets/ec6d98d7-a304-4d8d-8371-f8fc958bc7bf" />

yipfram commented:
Hi @JhumanJ  I'm not part of the staff, just a community member :)

JhumanJ commented:
@peaklabs-dev can you help here :) ?

ShadowArcanist commented:
> Hey @yipfram thanks for approving. Issue still have waiting for changes label, can we please get this changed? When we can expect this to be merged? Thanks <img alt="CleanShot 2025-11-08 at 13 09 20@2x" width="1188" height="160" src="https://private-user-images.githubusercontent.com/11312432/511670531-ec6d98d7-a304-4d8d-8371-f8fc958bc7bf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI2MTc0MzYsIm5iZiI6MTc2MjYxNzEzNiwicGF0aCI6Ii8xMTMxMjQzMi81MTE2NzA1MzEtZWM2ZDk4ZDctYTMwNC00ZDhkLTgzNzEtZjhmYzk1OGJjN2JmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA4VDE1NTIxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhMDYzNGQyYmU1MDQ0YWFjNWIzYTA2ZTY3ZjlkMTJjYTFiN2EyNjdmNjJmYWMxZmI2OTNkYzc5M2Q1YjM5NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.V1sgglNssBS1uSxNAOu4QQWxm-yWsfWXQUSl-Fttw3A">


Thank you for your submission. I wasn't able to approve the changes yet because the suggestion mentioned [here]( https://github.com/coollabsio/coolify/pull/5875#discussion_r2443342181 )
 has not been implemented. This suggestion ensures consistency in formatting the environment variables, in line with other templates in the repository. Once that change has been made, I will be happy to test the template and, assuming everything works as expected, mark the PR as ready to merge.

JhumanJ committed:
Update opnform.yaml to use version 1.10.2 for API and UI images, remove SELF_HOSTED environment variable, and adjust environment variable syntax for consistency

JhumanJ commented:
Thanks for getting back to us @ShadowArcanist, I have just updated the template again

JhumanJ commented:
Hate to be chasing @ShadowArcanist, but we'd really like to get this merged after all these months 🙏
I'm available if anything else needs to be done here - thanks!

JhumanJ committed:
Merge branch 'next' into add-opnform-template

ShadowArcanist commented:
> Hate to be chasing @ShadowArcanist, but we'd really like to get this merged after all these months 🙏 I'm available if anything else needs to be done here - thanks!

Hey, I will get back to this in next 48hrs, currently I have lot going on. The template looks good to me but I just have to test it on my dev coolify instance.

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,228 @@
+  APP_ENV: production
+  APP_KEY: ${SERVICE_BASE64_APIKEY}
+  APP_DEBUG: ${APP_DEBUG:-false}
+  APP_URL: ${SERVICE_FQDN_OPNFORM}
```
```suggestion
  APP_URL: ${SERVICE_FQDN_NGINX}
```

ShadowArcanist commented on templates/compose/opnform.yaml:
```diff
@@ -0,0 +1,228 @@
+  nginx:
+    image: nginx:1.29.2
+    environment:
+      - SERVICE_URL_OPNFORM
```
```suggestion
      - SERVICE_URL_NGINX
```

ShadowArcanist reviewed (changes requested):
Hey @JhumanJ I just tested this template on Coolify v4.0.0-beta.442
, it's working but one issue which is the form share url is wrong:

<img width="2055" height="1250" alt="Image" src="https://github.com/user-attachments/assets/21933b3e-0f28-4240-944c-f109326ac90b" />

After debugging for an hour I figured out that it works fine with the default Coolify generated domain, so I checked the env and found Coolify is not updating value of `SERVICE_FQDN_OPNFORM` when I change the domain for the service but it does update `SERVICE_URL_OPNFORM`.

I am not exactly sure if this is an bug on Coolify (I have to test this further), but replacing `SERVICE_FQDN_OPNFORM` and `SERVICE_URL_OPNFORM` with `SERVICE_FQDN_NGINX` and `SERVICE_URL_NGINX` fixes the issue:

<img width="2068" height="1119" alt="Image" src="https://github.com/user-attachments/assets/9804ed39-563e-42aa-a659-6a7b56f9ec3e" />

Please request a review after you made the suggested change (I just have to approve the PR and add a label)

JhumanJ committed:
Update templates/compose/opnform.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

JhumanJ committed:
Update templates/compose/opnform.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

JhumanJ committed:
fix(opnform): update APP_URL environment variable and remove unused nginx environment variable

JhumanJ commented:
Thank you so much @ShadowArcanist for your time!
I just updated the templates, replacing `SERVICE_FQDN_NGINX` with `SERVICE_URL_NGINX` which fixed the broken links when opening the form from the app. Also removed a variable that's not needed anymore on nginx. 
I ran some more tests changing the domain and it's working just fine.
Thanks again!

JhumanJ committed:
Merge branch 'next' into add-opnform-template

ShadowArcanist commented:
> Thank you so much @ShadowArcanist for your time! I just updated the templates, replacing `SERVICE_FQDN_NGINX` with `SERVICE_URL_NGINX` which fixed the broken links when opening the form from the app. Also removed a variable that's not needed anymore on nginx. I ran some more tests changing the domain and it's working just fine. Thanks again!

Keep the SERVICE_URL_NGINX under the nginx service, right now Coolify doesn't autognerate domain so users might think the template is broken:
<img width="684" height="1095" alt="image" src="https://github.com/user-attachments/assets/576eb619-8490-4497-8d0a-527c2c8630f0" />

It will show the urls if you are just editing the template via the edit compose feature but new deployments won't see the urls

JhumanJ committed:
feat(opnform): add SERVICE_URL_NGINX environment variable to nginx service

JhumanJ commented:
@ShadowArcanist should be ok now!

JhumanJ committed:
Upgrade opnform API and UI images to version 1.12.1 in opnform.yaml

JhumanJ commented:
Hey @ShadowArcanist I just bumped image version if you can please take a look 🙏

Also, @peaklabs-dev when can we hope to get this merged? Thanks!
